src/common/slice.rs

```diff
@@ -9,15 +9,19 @@ pub struct UncheckedSlice<T> {
 
 impl<T> UncheckedSlice<T> {
     #[inline]
-    unsafe fn get_unchecked(&self, index: usize) -> &T {
+    pub fn get_unchecked(&self, index: usize) -> &T {
```

Comment:
Is there any reason on make the function "safe" rather than moving the safe responsibility on the caller?

Reply:
If I don't do that, the unsafe annotation propagates. That is, the function that calls it needs to be annotated as unsafe and the function that calls that and  ....